apis/python/src/tiledbsoma/soma_object.cc

```diff
@@ -58,8 +58,11 @@ void load_soma_object(py::module& m) {
                std::optional<std::pair<uint64_t, uint64_t>> timestamp,
                std::optional<std::string> clib_type) -> py::object {
                 try {
-                    auto soma_obj = SOMAObject::open(
-                        uri, mode, context, timestamp, clib_type);
+                    auto soma_obj = ([&]() {
+                        py::gil_scoped_release release;
+                        return SOMAObject::open(
+                            uri, mode, context, timestamp, clib_type);
+                    })();
                     auto soma_obj_type = soma_obj->type();
```

Comment:
Was wondering if this is the same as doing

```
py::gil_scoped_release release;
auto soma_obj = SOMAObject::open(
                        uri, mode, context, timestamp, clib_type);
py::gil_scoped_acquire acquire;
```

Reply:
I'm not a pybind expert, but I believe it is the same (with a bit more overhead due to the need to destruct both objects).

I did the lambda because I wanted a protected scope, to ensure we didn't accidentally add any code that messed with python objects (I believe this was the bug in the previous PR).

So, TL;DR - it is mostly style, if I understand things correctly.